### PR TITLE
Update DLAMI Notebook server

### PR DESCRIPTION
**Pull request by kenoir**
### What is this PR trying to achieve?

Update the old data science EC2 instances to be more suited to their use case.

This change:
- Provides a persistent EBS volume that is automatically detached/attached
- Pins the version of the DLAMI AMI we use to "latest"
- Provides a static domain name to connect to: `notebook.wecodev.com`
- Removes the necessity to have multiple instance types (they can be updated or duplicated on demand).

```
pepperami: A single serve product with a mysterious but irresistible make up.
```

### Who is this change for?

@harrisonpim

### CODE REVIEW

File: data_science/terraform/modules/dlami/variables.tf [hunk 1]
```diff
@@ -1,62 +1,33 @@
-variable "name" {
-  description = "Name of the ASG to create"
-}
+variable "name" {}
+variable "instance_type" {}
 
-variable "subnet_list" {
-  type = "list"
+variable "hashed_password" {
+  default = "sha1:5310f21e370d:a4d66e725c179218638c21c03d83933aa066db2d"
```
Comment (hblanks) [inline data_science/terraform/modules/dlami/variables.tf]: Not sure, bate are these potentially creds in VCS? Maybe instead use TF's data kms_secret?

File: data_science/terraform/stack/notebook/main.tf [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+resource "aws_route53_record" "notebook" {
+  zone_id = "${aws_route53_zone.wecodev.zone_id}"
+  name    = "notebook.wecodev.com"
+  type    = "CNAME"
+  ttl     = "300"
+  records = ["${module.dlami.public_dns}"]
+}
+
+module "dlami" {
+  source = "../../modules/dlami"
```
Comment (hblanks) [inline data_science/terraform/stack/notebook/main.tf]: It feels slightly odd that this module is actually an EC2 instance not an AMI. But, just a thought there.

**Comment by jamesgorrie** [comment]
Pepperami is spelt peperami.

File: data_science/terraform/modules/dlami/main.tf [hunk 3]
```diff
@@ -36,44 +8,46 @@ module "security_groups" {
...
 module "instance_profile" {
   source = "git::https://github.com/wellcometrust/terraform.git//ec2/modules/instance_profile?ref=v11.7.2"
 
   name = "${var.name}"
 }
 
-data "template_file" "requirements" {
-  template = "${file("${path.module}/templates/requirements.txt")}"
-}
+resource "aws_instance" "data" {
```
Comment (jamesgorrie) [inline data_science/terraform/modules/dlami/main.tf]: datascience? Don't really mind.

File: data_science/terraform/modules/dlami/data.tf [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+data "aws_subnet_ids" "public" {
+  vpc_id = "${var.vpc_id}"
+
+  tags {
+    Availability = "public"
+  }
+}
+
+data "aws_subnet" "selected" {
```
Comment (jamesgorrie) [inline data_science/terraform/modules/dlami/data.tf]: What does `selected` mean?
Just confused me further down.

File: data_science/terraform/modules/dlami/variables.tf [hunk 5]
```diff
@@ -1,62 +1,33 @@
...
   description = "CIDR for SSH access to EC2 instances"
 }
-
-variable "custom_security_groups" {
-  type    = "list"
-  default = []
+variable "key_name" {}
+variable "ebs_volume_size" {
+  default = 250
 }
```
Comment (jamesgorrie) [inline data_science/terraform/modules/dlami/variables.tf]: Why have `vpc_id` and `key_name` variables?
Could make it frustrating to have to do each time.

**Pull request merged**